File: binance4j-market/src/main/java/com/binance4j/market/param/PriceTickersParams.java
package com.binance4j.market.param;

import java.util.List;
import java.util.stream.Collectors;

import com.binance4j.core.annotation.Param;
import com.binance4j.core.param.Params;
import com.binance4j.market.client.MarketClient;

/**
 * {@link MarketClient#getTicker} params.
 * 
 * @param symbols Symbols. Format: '["BTCBUSD","BNBBUSD"]'
 */
@Param(weight = 2, recvWindow = false, timestamp = false)
public record PriceTickersParams(String symbols) implements Params {

	/**
	 * Creates an instance of the class.
	 * 
	 * @param symbols Trading pair list.
	 */
	public PriceTickersParams(String symbols) {
		this.symbols = "[" + List.of(symbols.split(",")).stream().map(s -> String.format("\"%s\"", s.trim())).collect(Collectors.joining(",")) + "]";
	}

	/**
	 * Creates an instance of the class.
	 * 
	 * @param symbols Trading pair list.
	 */
	public PriceTickersParams(List<String> symbols) {
		this(symbols.stream().collect(Collectors.joining(",")));
	}
}

File: connectors/src/main/kotlin/com/binance4j/market/param/PriceTickersParams.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.binance4j.market.param

import com.binance4j.core.param.Params
import com.binance4j.market.client.MarketClient

/**
 * [MarketClient.getTicker] params.
 *
 * @param symbols Symbols. Format: '["BTCBUSD","BNBBUSD"]'
 */
data class PriceTickersParams(var symbols: String) : Params {
	override fun timestamp(): Long? = null
	override fun recvWindow(): Long? = null
	
	constructor (symbols: List<String>) : this("[" + symbols.joinToString(",") { s ->
		String.format(
			"\"%s\"",
			s.trim()
		)
	} + "]")
}